### DISCUSSION TITLE
Factory for UIKit navigation + SwiftUI

### DISCUSSION THREAD

**Discussion by betulaksufo** [hmlongco/Factory#80] [Q&A]
I'm a bit confused when using Factory. I use UIKit navigation and SwiftUI as I don't trust SwiftUI Navigation yet. However my container for each page looks like this:

```
extension Container {
    var exampleViewModel: Factory<ExampleViewModel> {
        self { ExampleViewModel() }.shared
    }

    var exampleView: Factory<ExampleView> {
        self { ExampleView(viewModel: self.exampleViewModel()) }.shared
    }

    var exampleCoordinator: Factory<ExampleCoordinator> {
        self { ExampleCoordinator() }.shared
    }

    var exampleNavigationController: Factory<UINavigationController> {
        self { UINavigationController() }.shared
    }

    var exampleViewController: Factory<UIViewController> {
        self {
            let viewController = UIHostingController(rootView: self.exampleView())
            return viewController
        }.shared
    }
}
```

Is it a good approach to create SwiftUI views with Factory? When I examine projects using Factory, I see that my approach is very different. Do you have any suggestions?

**Comment by hmlongco** [comment]
I would never create SwiftUI or UIKit views or controllers with Factory. Let SwiftUI or UIKit create the views or controllers as needed, and use Factory to provide them with the view models and services that they require.
